docs: misc

justinmk opened the pull request:
- docs: remove "#" comment char in `@return`

justinmk committed:
docs: misc

Co-authored-by: Kevin Pham <keevan.pham@gmail.com>

zeertzjq commented on MAINTAIN.md:
```diff
@@ -136,36 +136,38 @@ Non-technical dependencies
+      `-latest` tags here as it makes it difficult to determine from an
+      unrelated PR if a failure is due to the PR itself or due to GitHub bumping
+      the `-latest` tag without our knowledge. There's also a high risk that
+      automatic bumping the CI versions will fail due to manual work being
```
```suggestion
      automatically bumping the CI versions will fail due to manual work being
```

justinmk committed:
docs: remove "#" comment char in @return

Everything after a "#" char is a "description" comment, i.e. luals won't
treat it as a type, name, etc. But "#" should not be present in the
generated docs (such as :help docs).
https://github.com/LuaLS/lua-language-server/wiki/Annotations#return

justinmk commented on CONTRIBUTING.md:
```diff
@@ -289,7 +289,8 @@ For convenience you can filter the regeneration by target (api, lua, lsp) using
 
-Lua documentation uses a subset of [EmmyLua] annotations. See [:help dev-doc-lua][dev-doc-lua].
+Use [LuaLS] annotations in Lua docstrings to annotate parameter types, return
+types, etc. See [:help dev-doc-lua][dev-doc-lua].
```
we're using luals annotations, not emmylua, right? @clason

lewis6991 replied:
Well, emmylua isn't really active anymore so I think LuaLS is redefining it.

See: https://github.com/LuaLS/lua-language-server/issues/980#issuecomment-1082841463

lewis6991 commented:
I'm not sure this was a good change. `#` is used to deliminate the return type from the description. Without it, the first word is used as a name for the return value.

https://github.com/LuaLS/lua-language-server/wiki/Annotations#return

If you use LSP hover on a function, you can see the effect it has.
